File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        
        left = self.isValidBST(root.left)
        right = self.isValidBST(root.right)

        lc = root.left.val < root.val if root.left else True
        rc = root.right.val > root.val if root.right else True

        return left and right and lc and rc

File: Data Structures & Algorithms/populating-next-right-pointers-in-each-node/submission-1.js
/**
 * Definition for a binary tree node.
 * class Node {
 *     constructor(val = 0, left = null, right = null, next = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {Node} root
     * @return {Node}
     */

    // []
    // [1]
    // 1 -> [2, 3], 

    connect(root) {
        if(root == null){
            return root;
        }

        const q = new Queue();
        q.enqueue(root);

        while(q.size() > 0){
            const size = q.size();

            for(let i = 0; i < size; i++){
                const node = q.dequeue();
                
                if(i !== (size - 1)){
                    node.next = q.front();
                }

                if(node.left){
                    q.enqueue(node.left);
                }
                if(node.right){
                    q.enqueue(node.right);
                }
            }
        }


        return root;
        
        
    }
}